index.bs

```diff
@@ -6003,6 +6007,36 @@ leakage due to such an attack:
         to the preceding authentication step, where it may be easier to solve.
 
 
+## Privacy leak via credential IDs ## {#sctn-credential-id-privacy-leak}
+
+[INFORMATIVE]
+
+This privacy consideration applies to [=[RPS]=] supporting [=single-factor=] authentication with [=non-resident credentials=].
+
+In this case the {{PublicKeyCredentialRequestOptions/allowCredentials}} argument risks leaking [PII],
+if the user can initiate an [=authentication ceremony=] by only providing a username.
+[=Credential IDs=] are designed to not be correlatable between [=[RPS]=],
+but the length of a [=credential ID=] might be a hint as to what type of [=authenticator=] created it.
+It is likely that a user will use the same username and set of [=authenticators=] for several [=[RPS]=],
+so the number of [=credential IDs=] in {{PublicKeyCredentialRequestOptions/allowCredentials}} and their lengths
+might serve as a global correlation handle to de-anonymize the user.
```

Comment:
Perhaps we should recommend a 1 or 2 lengths for credential ids. That would reduce the slight leak.

Reply:
Even if we were to recommend that, I don't think it would have much effect in practice. There are already lots of authenticators in the field that cannot change, so we're still going to have to deal with different credential ID lengths for the foreseeable future.